qcodes/instrument_drivers/cryogenic/CryogenicSMS120C.py

```diff
@@ -0,0 +1,414 @@
+        if self._get_rampStatus() == 1:
+            log.error('Cannot switch heater during a ramp, first pause the ramp.')
+        else:
+            # Switch ON, if currently OFF
+            if val == 1 and (self._get_switchHeater() == False):
+                strHeaterStatus = self.ask('HEATER %d' % val)
+                switchHeater = 1
+                print('Waiting 30s for switch heater to warm up.')
+                time.sleep(30)  # wait for magnet to settle after switch
+            # Switch OFF, if currently ON
+            elif val == 0 and (self._get_switchHeater() == True):
+                strHeaterStatus = self.ask('HEATER %d' % val)
+                switchHeater = 0
+                if self._get_field != 0:  # condition for persistent mode
+                    self.persistent_mode = 1
+                    print('Waiting 60s for switch heater to cool.')
+                    # wait for magnet to settle into persistent mode after
+                    # switch
+                    time.sleep(60)
+                else:
+                    print('Waiting 30s for switch heater to cool.')
+                    time.sleep(30)  # wait for magnet to settle after switch
+            else:  # assume no change to current switch heater state
+                strHeaterStatus = self.ask('HEATER %d' % val)
+                log.info(strHeaterStatus)
+            return switchHeater
+
+    def _set_pauseRamp(self, val):  # Pause magnet controller Pause=1, Unpause=0
+        self.ask('PAUSE %d' % val)
+
+    # Set ramp speed Amps/sec , check it is reasonable if it is being manually
+    # modified
+    def _set_rampRate(self, val):
+        if self._current_ramp_limit is None:
+            self._current_ramp_limit = CryogenicSMS120C.default_current_ramp_limit
+
+        if val <= self._current_ramp_limit:
+            self.ask('SET RAMP %0.2f' % val)
+            return True
+        elif val > CryogenicSMS120C.default_max_current_ramp_limit:
+            self.ask('SET RAMP %0.2f' %
+                     CryogenicSMS120C.default_current_ramp_limit)
+            msg = 'Requested rate of {} is unsafe and over the maximum limit of {} A/s. Coerced to default ramp rate.'
+            log.error(msg.format(
+                val, CryogenicSMS120C.default_max_current_ramp_limit))
+            return False
+        else:
+            msg = 'Requested ramp speed is over the limit of {} A/s. Change limit, at your own risk after consulting the SMS120C manual.'
+            log.error(msg.format(self._current_ramp_limit))
+            return False
+
+    # Check magnet state and ramp speed to see if it is safe to ramp, returns
+    # boolean
+    def _can_startRamping(self):
+        state = self._get_rampStatus()
+
+        if self._get_rampRate() <= self._current_ramp_limit:
+            if state == 2:         # Quench
+                log.error(
+                    __name__ + ': Magnet quench detected - please check magnet status before ramping.')
```

Comment:
SInce you are now using module level logging (`log = logging.getLogger(__name__)`), there is no reason to prepend the `__name__`.